app-backend/tile/src/main/scala/routes/MosaicRoutes.scala

```diff
@@ -50,15 +52,24 @@ object MosaicRoutes extends LazyLogging {
               val poly = Projected(Extent.fromString(bbox).toPolygon(), 4326)
                 .reproject(LatLng, WebMercator)(3857)
               val extent = poly.envelope
-              acceptContentType match {
+              val future = acceptContentType match {
                 case Some("image/tiff") => mosaic
                     .map{ m => MultibandGeoTiff(m, extent, WebMercator) }
                     .map(tiffAsHttpResponse)
                     .value
                 case _ => mosaic.map(_.renderPng)
                     .map(pngAsHttpResponse)
                     .value
+                  
               }
+
+              future onComplete {
+                case Success(s) => s
+                case Failure(e) =>
+                  logger.error(s"Message: ${e.getMessage}\nStack trace: ${RfStackTrace(e)}")
```

Comment:
I don't think `onComplete` is mutating so the value of `future` would remain intact.